src/trans.rs

```diff
@@ -314,12 +314,12 @@ impl<'v, 'tcx: 'v> BinaryenFnCtxt<'v, 'tcx> {
                     let a = self.trans_operand(a);
                     let b = self.trans_operand(b);
                     let op = match *op {
-                        BinOp::Add => BinaryenAdd(),
-                        BinOp::Sub => BinaryenSub(),
-                        BinOp::Mul => BinaryenMul(),
-                        BinOp::Div => BinaryenDivS(),
-                        BinOp::Eq => BinaryenEq(),
-                        BinOp::Ne => BinaryenNe(),
+                        BinOp::Add => BinaryenAddInt32(),
```

Comment:
Yeah I agree, and this is why I was asking the question of the original behavior I wanted to preserve. I also saw that trans_operand panics if numerical constants were not i32 and those made me think this was ok for now but should definitely be dealt with as more of the numeric types are handled. assert + comment mentioning the TODO could be worthwhile right ?